javascript-practice/src/js/services/userService.js

```diff
@@ -48,14 +37,37 @@ export default class UserService extends CommonService {
    * @returns {Object || null} Return new User Object if find, otherwise return null.
    */
   async getUserByEmail(email) {
-    const id = await this.getUserIdByEmail(email);
-
-    if (id) {
-      const user = await this.getDataFromId(id);
+    const result = await this.getDataFromProp('email', email);
 
-      return new User(user);
+    if (result) {
+      return result;
     }
 
     return null;
   }
+
+  async validateUser(email, password) {
+    const user = await this.getUserByEmail(email);
+
+    if (user) {
```

Comment:
This code is too complicated, pls try to refactor it

Reply:
Fixed.